metadata-ingestion/src/datahub/ingestion/source/sac/sac.py

```diff
@@ -0,0 +1,808 @@
+
+                schema_field = SchemaFieldClass(
+                    fieldPath=column.name,
+                    type=self.get_schema_field_data_type(
+                        column.property_type, column.data_type
+                    ),
+                    nativeDataType=native_data_type,
+                    description=column.description,
+                    isPartOfKey=column.is_key,
+                )
+
+                schema_fields.append(schema_field)
+
+                if column.is_key:
+                    primary_fields.append(column.name)
+
+            mcp = MetadataChangeProposalWrapper(
+                entityType="dataset",
+                changeType=ChangeTypeClass.UPSERT,
+                entityUrn=dataset_urn,
+                aspectName="schemaMetadata",
+                aspect=SchemaMetadataClass(
+                    schemaName=model.model_id,
+                    platform=make_data_platform_urn(self.platform),
+                    version=0,
+                    hash="",
+                    platformSchema=SchemalessClass(),
+                    fields=schema_fields,
+                    primaryKeys=primary_fields,
+                ),
+            )
+
+            yield MetadataWorkUnit(
+                id=f"dataset-upstream-lineage-{dataset_urn}", mcp=mcp
+            )
+
+        if model.system_type in ("BW", "HANA") and model.external_id is not None:
+            upstream_dataset_name: Optional[str] = None
+
+            if model.system_type == "BW" and model.external_id.startswith(
+                "query:"
+            ):  # query:[][][query]
+                query = model.external_id[11:-1]
+                upstream_dataset_name = self.get_query_name(query)
+            elif model.system_type == "HANA" and model.external_id.startswith(
+                "view:"
+            ):  # view:[schema][schema.namespace][view]
+                schema, namespace_with_schema, view = model.external_id.split("][", 2)
+                schema = schema[6:]
+                namespace: Optional[str] = None
+                if len(schema) < len(namespace_with_schema):
+                    namespace = namespace_with_schema[len(f"{schema}.") :]
+                view = view[:-1]
+                upstream_dataset_name = self.get_view_name(schema, namespace, view)
+
+            if upstream_dataset_name is not None:
+                if model.connection_id in self.config.connection_mapping:
+                    connection = self.config.connection_mapping[model.connection_id]
+                    platform = (
+                        connection.platform
+                        if connection.platform
+                        else model.system_type.lower()
+                    )
+                    platform_instance = connection.platform_instance
+                    env = connection.env
+                else:
+                    platform = model.system_type.lower()
```

Comment:
does it make sense to print some debug logs here? e.g. no connection_mapping found, doing best-effort inference

or are the logs below sufficient?

Reply:
I have added a log with info level (I think debug is not enough?) - besides the log message you would notice that the best-effort approach was used, because there will be platform instances which are named after the connections in SAC...